app/controllers.rb

```diff
@@ -56,19 +57,22 @@
   end
 end
 
+# TODO: Use of non-underscore-prefixed option parameters is still
+# supported but deprecated, and should be removed at some point soon
+SUPPORT_DEPRECATED_OPTION_SYNTAX = true
 def get_search_args_from_params(params)
-  endpoint = params.delete("endpoint")
   options = {
-    api: endpoint,
-    sort: params.delete("sort"),
-    zip: params.delete("zip"),
-    distance: params.delete("distance"),
-    fields: (params.delete('fields') || "").split(','),
-    per_page: params.delete("per_page") || DataMagic.config.page_size,
-    page: params.delete("page").to_i || 0
+    endpoint: params.delete("endpoint"),
+    format:   params.delete("format")
   }
-  format = params.delete('format')
-  [endpoint, options, format]
+  %w(sort fields zip distance page per_page).each do |opt|
```

Comment:
if SUPPORT_DEPRECATED_OPTION_SYNTAX  would make this a bit more readable (and efficient)